File: realm_oc_demo/realm_oc_demo/ViewController.m
//
//  ViewController.m
//  realm_oc_demo
//
//  Created by allison on 2021/1/22.
//

#import "ViewController.h"
#import <Realm/Realm.h>
#import "MigrationModel.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [self setDefaultRealmForUser:@"zhangsan"];
    [self migrationAction];
}


///数据库迁移
- (void)migrationAction {
    NSLog(@"数据库路径:%@",[[RLMRealmConfiguration defaultConfiguration]fileURL]);
    MigrationModel *model = [[MigrationModel alloc]init];
    model.name = @"allison";
    model.age = 18;
    model.fullName = @"allison-fullname";
    model.fullName2 = @"fullname22222";
    model.fullName3 = @"fullname33333";

    RLMRealm *realm = [RLMRealm defaultRealm];
    [realm transactionWithBlock:^{
        [realm addObject:model];
    }];
    
}

- (void)setDefaultRealmForUser:(NSString *)userName {
//    RLMRealm *realm = [RLMRealm defaultRealm];
    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
    //使用默认的目录，但是使用用户名来替换默认的文件名
    config.fileURL = [[[config.fileURL URLByDeletingLastPathComponent]URLByAppendingPathComponent:userName]URLByAppendingPathExtension:@"realm"];
    //将这个配置应用到默认的realm数据库中
    [RLMRealmConfiguration setDefaultConfiguration:config];
    
}

@end

File: realm_swift_demo/realm_demo/ViewController.swift
//
//  ViewController.swift
//  realm_demo
//
//  Created by allison on 2021/1/17.
//

import UIKit
import RealmSwift

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
      
        print("数据库path:",Realm.Configuration.defaultConfiguration.fileURL)
        
        let model = MigrationModel()
        model.name = "allison"
        model.age = 18
        model.fullName1 = "fullName001"
        model.fullName2 = "fullName002"

        let realm = try! Realm()
        try! realm.write {
            realm.add(model)
        }
        
    }
    

}


